ci/ReleaseTool/Common.cs

```diff
@@ -55,5 +57,108 @@ private static string AppendDirectorySeparator(string originalPath)
 
             return originalPath + directorySeparator;
         }
+
+        public static bool IsMarkdownHeading(string markdownLine, int level, string startTitle = null)
+        {
+            var heading = $"{new string('#', level)} {startTitle ?? string.Empty}";
+
+            return markdownLine.StartsWith(heading);
+        }
+
+        public static bool UpdateChangeLog(string ChangeLogFilePath, string Version, GitClient gitClient, string ChangeLogReleaseHeadingTemplate)
+        {
+            using (new WorkingDirectoryScope(gitClient.RepositoryPath))
+            {
+                if (File.Exists(ChangeLogFilePath))
+                {
+                    var originalContents = File.ReadAllText(ChangeLogFilePath);
+                    var changelog = File.ReadAllLines(ChangeLogFilePath).ToList();
+                    var releaseHeading = string.Format(ChangeLogReleaseHeadingTemplate, Version,
+                        DateTime.Now);
+                    var releaseIndex = changelog.FindIndex(line => IsMarkdownHeading(line, 2, $"[`{Version}`] - "));
+                    // If we already have a changelog entry for this release, replace it.
+                    if (releaseIndex != -1)
+                    {
+                        changelog[releaseIndex] = releaseHeading;
+                    }
+                    else
+                    {
+                        // Add the new release heading under the "## Unreleased" one.
+                        // Assuming that this is the first heading.
+                        var unreleasedIndex = changelog.FindIndex(line => IsMarkdownHeading(line, 2));
```

Comment:
So this is making the assumption that the first # MarkDown Heading will be Unreleased? Why not just include a string to make sure.

Reply:
I think this is searching through all lines, looking for the first ## heading, and adding new things under that.
We could make it search for Unreleased specifically, I guess.

Reply:
So, I guess is we are doing it or we are not? xD

Reply:
I don't think I want to do that, as I feel like the release tool should handle failures / failed expectations as gracefully as possible. So if we made it search for "Unreleased" - I feel like that's too pedantic. And the changelog has to be humanly reviewed after this point anyway. I guess we could print a warning if the line is different from what is expected?